lib/ex_webrtc/rtp_receiver/nack_generator.ex

```diff
@@ -0,0 +1,110 @@
+defmodule ExWebRTC.RTPReceiver.NACKGenerator do
+  @moduledoc nil
+  # for now, it mimics the Pion implementation, but there's some issues and remarks
+  # 1) NACKs are send at constant interval
+  # 2) no timing rules (like rtt) are taken into account
+  # 3) NACKs will be sent until max_nacks NACKs has been sent or sn is older than @max_sn
+
+  alias ExRTCP.Packet.TransportFeedback.NACK
+
+  @max_nack 3
+  @max_sn 0xFFFF
+  @breakpoint 0x7FFF
+
+  @type t() :: %__MODULE__{
+          media_ssrc: non_neg_integer() | nil,
+          sender_ssrc: non_neg_integer(),
+          lost_packets: %{non_neg_integer() => non_neg_integer()},
+          last_sn: non_neg_integer() | nil,
+          max_nack: non_neg_integer()
+        }
+
+  defstruct sender_ssrc: 1,
+            media_ssrc: nil,
+            lost_packets: %{},
+            last_sn: nil,
+            max_nack: @max_nack
+
+  @doc """
+  Records incoming RTP Packet.
+  """
+  @spec record_packet(t(), ExRTP.Packet.t()) :: t()
+  def record_packet(generator, packet)
+
+  def record_packet(%{last_sn: nil} = generator, packet) do
+    %__MODULE__{generator | media_ssrc: packet.ssrc, last_sn: packet.sequence_number}
+  end
+
+  def record_packet(generator, packet) do
+    %__MODULE__{
+      lost_packets: lost_packets,
+      last_sn: last_sn,
+      max_nack: max_nack
+    } = generator
+
+    delta = packet.sequence_number - last_sn
+    in_order? = delta < -@breakpoint or (delta > 0 and delta < @breakpoint)
```

Comment:
Maybe we could move this to a separate RTP utils module

Reply:
I wouldn't, the amount of repetition is nearly non-existent and moving a single-line function to a separate module would make it harder to read, arguably.